mParticle-Apptimize/MPKitApptimize.m

```diff
@@ -0,0 +1,133 @@
+    
+    _configuration = configuration;
+    
+    if (startImmediately) {
+        [self start];
+    }
+    
+    return self;
+}
+
+- (void)start {
+    static dispatch_once_t kitPredicate;
+    
+    dispatch_once(&kitPredicate, ^{
+        [Apptimize startApptimizeWithApplicationKey:self.configuration[@"appkey"]];
+        
+        _started = YES;
+        
+        dispatch_async(dispatch_get_main_queue(), ^{
+            NSDictionary *userInfo = @{mParticleKitInstanceKey:[[self class] kitCode]};
+            
+            [[NSNotificationCenter defaultCenter] postNotificationName:mParticleKitDidBecomeActiveNotification
+                                                                object:nil
+                                                              userInfo:userInfo];
+        });
+    });
+}
+
+#pragma mark User attributes and identities
+
+- (MPKitExecStatus *)setUserAttribute:(NSString *)key value:(NSString *)value {
+    [Apptimize setUserAttributeString:value forKey:key];
+    
+    MPKitExecStatus *execStatus = [[MPKitExecStatus alloc] initWithSDKCode:@(MPKitInstanceApptimize) returnCode:MPKitReturnCodeSuccess];
+    return execStatus;
+}
+
+- (MPKitExecStatus *)removeUserAttribute:(NSString *)key {
+    [Apptimize removeUserAttributeForKey:key];
+    
+    MPKitExecStatus *execStatus = [[MPKitExecStatus alloc] initWithSDKCode:@(MPKitInstanceApptimize) returnCode:MPKitReturnCodeSuccess];
+    return execStatus;
+}
+
+ - (MPKitExecStatus *)setUserIdentity:(NSString *)identityString identityType:(MPUserIdentity)identityType {
+     if (identityType == MPUserIdentityCustomerId) {
+         [Apptimize setUserAttributeString:identityString forKey:CUSTOMER_ID_KEY];
+     } else if (identityType == MPUserIdentityCustomerId) {
+         [Apptimize setUserAttributeString:identityString forKey:ALIAS_KEY];
+     } else {
+         // didn't do anything
+         MPKitExecStatus *execStatus = [[MPKitExecStatus alloc] initWithSDKCode:@(MPKitInstanceApptimize) returnCode:MPKitReturnCodeUnavailable];
+         return execStatus;
+     }
+     
+     
+      MPKitExecStatus *execStatus = [[MPKitExecStatus alloc] initWithSDKCode:@(MPKitInstanceApptimize) returnCode:MPKitReturnCodeSuccess];
+      return execStatus;
+ }
+
+#pragma mark Events
+
+- (MPKitExecStatus *)logEvent:(MPEvent *)event {
+    [Apptimize track:event.name];
+    
+    MPKitExecStatus *execStatus = [[MPKitExecStatus alloc] initWithSDKCode:@(MPKitInstanceApptimize) returnCode:MPKitReturnCodeSuccess];
+    return execStatus;
+}
+
+- (MPKitExecStatus *)logScreen:(MPEvent *)event {
+    NSString *screenEvent = [NSString stringWithFormat:VIEWED_EVENT_FORMAT, event.name];
+    [Apptimize track:screenEvent];
+    
+    MPKitExecStatus *execStatus = [[MPKitExecStatus alloc] initWithSDKCode:@(MPKitInstanceApptimize) returnCode:MPKitReturnCodeSuccess];
+    return execStatus;
+}
+
+#pragma mark Assorted
+- (MPKitExecStatus *)setOptOut:(BOOL)optOut {
+    [Apptimize disable];
```

Comment:
Correct, I believe that would be way to go. If `optOut == true` then disable Apptimize, otherwise do nothing. The return exec status would remain the same, since the method was executed successfully, just no operation took place.